core/spec/models/spree/address_spec.rb

```diff
@@ -32,6 +32,7 @@
       expect(cloned.state_id).to eq(original.state_id)
       expect(cloned.state_name).to eq(original.state_name)
       expect(cloned.zipcode).to eq(original.zipcode)
+      expect(cloned.user_id).to eq(original.user_id)
```

Comment:
@MatthewKennedy actually, there is catch to this - we're cloning addresses from user to order when the user completes the checkout. Those cloned addresses shouldn't be associated with the user anymore - there should be only associated to order

Reply:
We may need a different solution then.

If you edit an address that’s not editable because it’s tied to an existing order, the clone happens to keep the address for the completed order, but upon save the newly cloned address is not accessible to the user.

Video of this prop to this change

https://youtu.be/a5GbSMAYHbs

Reply:
@damianlegawiec I've been looking at this over the past few hours, and as far as I can tell, the address should be associated with a user after checkout so that the address they just entered at checkout is then available in the user accounts section.

The clone attributed that I have changed in this PR, comes into play if the user then tries to edit an address that is already associated with an existing order, at this point the address from the order is cloned and presented to the user as an edit, even though technically its a new address.

This preserves the address id and details from the original address for the order details.

This video shows both the bug and the fix in action, don't know if it helps.
https://youtu.be/ix3HB7La4Rs

Reply:
@damianlegawiec If you like, I'll pull the change out of this PR?

Reply:
I've pulled this change out as it can be dealt with in its own PR.